[bitnami/redis] preExecsCmd type conflict

### Name and Version

bitnami/redis 20.2.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. Create a Chart depending on https://charts.bitnami.com/bitnami version 20.2.0 with the name redis
2. Use values.yaml below and run `helm lint`
3. See error `expected string; got []interface {}`
```
[ERROR] templates/: template: myproject/charts/redis/templates/replicas/application.yaml:50:36: executing "myproject/charts/redis/templates/replicas/application.yaml" at <include (print $.Template.BasePath "/scripts-configmap.yaml") .>: error calling include: template: myproject/charts/redis/templates/scripts-configmap.yaml:649:44: executing "myproject/charts/redis/templates/scripts-configmap.yaml" at <4>: wrong type for value; expected string; got []interface {}
```

Trying it with the second values_string.yaml would get the error
```
[ERROR] templates/: values don't meet the specifications of the schema(s) in the following chart(s):
redis:
- master.preExecCmds: Invalid type. Expected: array, given: string
```

### Are you using any custom parameters or values?

values.yaml
```yaml
redis:
  master:
    preExecCmds: ["redis-cli flushdb"]
```

values_string.yaml
```yaml
redis:
  master:
    preExecCmds: "redis-cli flushdb"
```

### What is the expected behavior?

Get at least one of the values.yaml to pass, either with a string or with an array of strings.

### What do you see instead?

Validation errors. This prevent any usage of non empty `preExecsCmd` after 20.0.4+

### Additional information

This was introduced in #29106 that enforces a schema for an array while the code uses it with `nindent` which works on strings.

--- a/bitnami/redis/CHANGELOG.md
+++ b/bitnami/redis/CHANGELOG.md
@@ -1,8 +1,12 @@
 # Changelog
 
-## 20.2.0 (2024-10-04)
+## 20.2.1 (2024-10-15)
 
-* [bitnami/redis] add extraPodSpec ([#29725](https://github.com/bitnami/charts/pull/29725))
+* [bitnami/redis] Fix preExecCmds parameter ([#29898](https://github.com/bitnami/charts/pull/29898))
+
+## 20.2.0 (2024-10-09)
+
+* [bitnami/redis] add extraPodSpec (#29725) ([0d2b826](https://github.com/bitnami/charts/commit/0d2b8269e0e5c1a287fcd8e44c623afa6311e761)), closes [#29725](https://github.com/bitnami/charts/issues/29725)
 
 ## <small>20.1.7 (2024-10-02)</small>
 

--- a/bitnami/redis/Chart.yaml
+++ b/bitnami/redis/Chart.yaml
@@ -36,4 +36,4 @@ maintainers:
 name: redis
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/redis
-version: 20.2.0
+version: 20.2.1

--- a/bitnami/redis/templates/scripts-configmap.yaml
+++ b/bitnami/redis/templates/scripts-configmap.yaml
@@ -232,7 +232,9 @@ data:
     {{- end }}
 
     {{- if .Values.replica.preExecCmds }}
-    {{- .Values.replica.preExecCmds | nindent 4 }}
+    {{- range $command := .Values.replica.preExecCmds }}
+    {{- $command | nindent 4 }}
+    {{- end }}
     {{- end }}
 
     {{- if .Values.replica.command }}
@@ -440,7 +442,9 @@ data:
     {{- end }}
     {{- end }}
     {{- if .Values.sentinel.preExecCmds }}
-    {{ .Values.sentinel.preExecCmds | nindent 4 }}
+    {{- range $command := .Values.sentinel.preExecCmds }}
+    {{- $command | nindent 4 }}
+    {{- end }}
     {{- end }}
     mv /opt/bitnami/redis-sentinel/etc/prepare-sentinel.conf /opt/bitnami/redis-sentinel/etc/sentinel.conf
     exec redis-server /opt/bitnami/redis-sentinel/etc/sentinel.conf {{- if .Values.tls.enabled }} "${ARGS[@]}" {{- end }} --sentinel
@@ -646,7 +650,9 @@ data:
     {{- end }}
     {{- end }}
     {{- if .Values.master.preExecCmds }}
-    {{ .Values.master.preExecCmds | nindent 4 }}
+    {{- range $command := .Values.master.preExecCmds }}
+    {{- $command | nindent 4 }}
+    {{- end }}
     {{- end }}
     {{- if .Values.master.command }}
     exec {{ .Values.master.command }} "${ARGS[@]}"
@@ -754,8 +760,9 @@ data:
     {{- end }}
     {{- end }}
     {{- if .Values.replica.preExecCmds }}
-    {{ .Values.replica.preExecCmds | nindent 4 }}
-    {{- end }}
+    {{- range $command := .Values.replica.preExecCmds }}
+    {{- $command | nindent 4 }}
+    {{- end }}    {{- end }}
     {{- if .Values.replica.command }}
     exec {{ .Values.replica.command }} "${ARGS[@]}"
     {{- else }}

